sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc

```diff
@@ -0,0 +1,296 @@
+
+== Kernel Properties
+
+```c++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+enum class streaming_interface_options_enum {
+  accept_downstream_stall,
+  remove_downstream_stall
+};
+
+enum class register_map_interface_options_enum {
+  wait_for_done_read,
+  do_not_wait_for_done_read
+};
+
+struct initiation_interval_key {
+  template <int initiation_interval_value>
+  using value_t = property_value<initiation_interval_key,
+    std::integral_constant<int, initiation_interval_value>>;
+};
+
+struct streaming_interface_key {
+  template <streaming_interface_options_enum option>
+  using value_t = property_value<streaming_interface_key,
+    std::integral_constant<streaming_interface_options_enum, option>>;
+};
+
+struct register_map_interface_key {
+  template <register_map_interface_options_enum option>
+  using value_t = property_value<register_map_interface_key,
+    std::integral_constant<register_map_interface_options_enum, option>>;
+};
+
+struct pipelined_key {
+  template <bool pipeline_directive, typename initiation_interval>
+  using value_t = property_value<pipelined_key, pipeline_directive,
+   initiation_interval>;
+};
+
+template <streaming_interface_options_enum option>
+inline constexpr streaming_interface_key::value_t<option> streaming_interface;
+
+inline constexpr streaming_interface_key::value_t<
+  streaming_interface_options_enum::accept_downstream_stall>
+    streaming_interface_accept_downstream_stall;
+
+inline constexpr streaming_interface_key::value_t<
+  streaming_interface_options_enum::remove_downstream_stall>
+    streaming_interface_remove_downstream_stall;
+
+inline constexpr register_map_interface_key::value_t<
+  register_map_interface_options_enum::wait_for_done_read>
+    register_map_interface_wait_for_done_read;
+
+inline constexpr register_map_interface_key::value_t<
+  register_map_interface_options_enum::do_not_wait_for_done_read>
+    register_map_interface_do_not_wait_for_done_read;
+
+inline constexpr register_map_interface_key::value_t<option>
+  register_map_interface;
+
+template<int initiation_interval_value>
+inline constexpr initiation_interval_key::value_t<initiation_interval_value>
+  initiation_interval;
+
+template<bool pipeline_directive, typename initiation_interval>
+inline constexpr pipelined_key::value_t<pipeline_directive,
+  initiation_interval> pipelined;
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+|===
+|Property|Description
+|`streaming_interface`
+|The `streaming_interface` property adds the requirement that the kernel must
+ have dedicated ports for input / output signals. This applies for both
+ control, and kernel argument data signals. The following values are supported:
+
+ * `accept_downstream_stall`: Provides the compiler the guarantee that
+    downstream logic will back-pressure the kernel.
+
+ * `remove_downstream_stall`: Provides the compiler the guarantee that
```

Comment:
It goes beyond that, right?  This directs the compiler to omit the port entirely.